lib/stable-asset/src/lib.rs

```diff
@@ -1132,23 +1068,22 @@ impl<T: Config> StableAsset for Pallet<T> {
 				let fee_recipient = pool_info.fee_recipient.clone();
 				T::Assets::mint_into(pool_info.pool_asset, &fee_recipient, fee_amount)?;
 				Self::deposit_event(Event::FeeCollected(who.clone(), pool_id, fee_recipient, fee_amount));
-				*maybe_pool_info = Some(PoolInfo {
-					pool_asset: pool_info.pool_asset,
-					assets: pool_info.assets,
-					precisions: pool_info.precisions,
-					mint_fee: pool_info.mint_fee,
-					swap_fee: pool_info.swap_fee,
-					redeem_fee: pool_info.redeem_fee,
-					total_supply,
-					a: pool_info.a,
-					balances,
-					fee_recipient: pool_info.fee_recipient,
-					account_id: pool_info.account_id,
-					pallet_id: pool_info.pallet_id,
-				});
+				pool_info.total_supply = total_supply;
+				pool_info.balances = balances;
 			}
 			Ok(())
 		})?;
 		Ok(().into())
 	}
+
+	fn modify_a(who: &Self::AccountId, pool_id: PoolId, a: Self::AtLeast64BitUnsigned) -> DispatchResult {
+		Pools::<T>::try_mutate_exists(pool_id, |maybe_pool_info| -> DispatchResult {
+			let pool_info = maybe_pool_info.as_mut().ok_or(Error::<T>::PoolNotFound)?;
+			ensure!(pool_info.admin == who.clone(), Error::<T>::NotAdmin);
```

Comment:
Anyone can create a pool and modify parameter A, which is not safe. Maybe can refer to Acala dex.

https://github.com/AcalaNetwork/Acala/blob/71a0b66be47896c023e33feb112c19c3b90cf5ee/modules/dex/src/lib.rs#L471

Reply:
Parameter A is best to change slowly in `on_initialize`

https://github.com/curvefi/curve-contract/blob/b0bbf77f8f93c9c5f4e415bce9cd71f0cdee960e/contracts/pools/3pool/StableSwap3Pool.vy#L149-L167

Reply:
I think pool creation should be permissionless, and only the admin(the contract creator) can modify the A value here.

Reply:
That will cause the pool to have a lot of useless data.

Modifying the value of A by the owner of the pool will generate arbitrage opportunities and cause users to lose money.

Reply:
I don't think there should be a global admin because we want that anyone in this chain can create their own pools. It's the owner's responsibility to check if such modification of `A` value is acceptable, not like there is an admin of the entire ecosystem. 
How badly you want us to support modifying `A` value? If it's not urgent, we can remove this feature for now.

Reply:
@xlc
Do we need permission control to create a pool?

Reply:
All the storage usage on chain must come with deposit or permissioned. So it needs to be either taking a deposit to make a pool (and refund when pool is destroyed), or only allow a configurable origin to create a pool.

We want to take the later approach in this case to matching behavior of Acala Swap

https://github.com/AcalaNetwork/Acala/blob/9870c765d133d64cc350789b78e3f098b27eaba3/modules/dex/src/lib.rs#L125-L126

Reply:
I see.

Reply:
Fixed